### PR TITLE
Web interface Quality of Life improvements

### PR DESCRIPTION
**Pull request by MashaNemt**
Fixes https://github.com/prometheus/prometheus/issues/4599 and introduces a small QoL improvement.

1. Resize text-input area on Graph page not only on text input, but on mouseclick as well. It helps to avoid tediously inputting something in the fields of interest on page reload;
2. Remember "All"/"Unhealthy" choice on target-overview when reloading page.

### CODE REVIEW

**Comment by zuzzas** [comment]
@juliusv, care to take a look? Seems really useful.

File: web/ui/static/js/graph/index.js [hunk 1]
```diff
@@ -62,21 +62,26 @@ Prometheus.Graph.prototype.initialize = function() {
...
-
-    // Auto-resize the text area on input.
-    var offset = this.offsetHeight - this.clientHeight;
-    var resizeTextarea = function(el) {
-        $(el).css('height', 'auto').css('height', el.scrollHeight + offset);
-    };
     $(this).on('keyup input', function() { resizeTextarea(this); });
   });
+
+  self.expr.click(function(e) {
```
Comment (juliusv) [inline web/ui/static/js/graph/index.js]: Ha funny, I also just worked on this today in the rewrite of this UI in React that I'm doing. But there it works to just bind on the `input` event alone to catch any changes to the field, then I think the click handler wouldn't be needed anymore in the future. But good for now.

File: web/ui/static/js/targets.js [hunk 2]
```diff
@@ -21,6 +21,15 @@ function showUnhealthy(_, container) {
 }
 
 function init() {
+  if (!localStorage.selectedTab || localStorage.selectedTab == "all-targets"){
+    $("div label:nth-child(1)").addClass("active");
+    $(".table-container").each(showAll);
+  }
+  else if (localStorage.selectedTab == "unhealthy-targets") {
```
Comment (juliusv) [inline web/ui/static/js/targets.js]: Style-wise, could you put this on the same line as the brace above it? (`} else if (...`).

File: web/ui/static/js/targets.js [hunk 3]
```diff
@@ -21,6 +21,15 @@ function showUnhealthy(_, container) {
 }
 
 function init() {
+  if (!localStorage.selectedTab || localStorage.selectedTab == "all-targets"){
+    $("div label:nth-child(1)").addClass("active");
```
Comment (juliusv) [inline web/ui/static/js/targets.js]: Maybe slightly better to find the labels by ID (`all-targets` / `unhealthy-targets`) and set the parent active?

Comment (MashaNemt) [reply web/ui/static/js/targets.js]: Done.

File: web/ui/static/js/targets.js (hunk 2 above)
Comment (MashaNemt) [reply web/ui/static/js/targets.js]: Done.

File: web/ui/static/js/graph/index.js (hunk 1 above)
Comment (MashaNemt) [reply web/ui/static/js/graph/index.js]: Looks like there is nothing for me to do then. :)

**Comment by juliusv** [comment]
:+1: Thanks!

**Pull request merged**